lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -576,21 +583,21 @@ newDBLayerWith _cacheBehavior tr ti SqliteContext{runQuery} = do
             let heights = Set.fromList $ sparseCheckpoints
                     (defaultSparseCheckpointsConfig epochStability)
                     (tip ^. #blockHeight)
-            modifyDBMaybe checkpointsVar $ \ws ->
+            modifyDBMaybe walletsVar $ \ws ->
                 case Map.lookup wid ws of
                     Nothing  -> (Nothing, ())
-                    Just cps ->
-                        let willKeep cp =
-                                (cp ^. #currentTip ^. #blockHeight ^. #getQuantity)
-                                `Set.member` heights
-                            slots = Map.filter willKeep (cps ^. #checkpoints)
-                        in  (Just $ Adjust wid $ RestrictTo $ Map.keys slots, ())
+                    Just wal ->
+                        let willKeep cp = getBlockHeight cp `Set.member` heights
+                            slots = Map.filter willKeep (wal ^. #checkpoints ^. #checkpoints)
+                            delta = Just $ Adjust wid $ (:[]) $ UpdateCheckpoints $
```

Comment:
(minor stylistic remarks) What about `Just . Adjust wid . (:[]) . UpdateCheckpoints . RestrictTo $ Map.keys slots` ?

Reply:
I agree. Actually, I decided that I do not like `(:[])` and added a commit where I use `[ x ]` instead.